src/jit/lower.cpp

```diff
@@ -147,18 +146,77 @@ GenTree* Lowering::LowerNode(GenTree* node)
     {
         case GT_IND:
             TryCreateAddrMode(LIR::Use(BlockRange(), &node->gtOp.gtOp1, node), true);
+            ContainCheckIndir(node->AsIndir());
             break;
 
         case GT_STOREIND:
-            LowerStoreInd(node);
+            TryCreateAddrMode(LIR::Use(BlockRange(), &node->gtOp.gtOp1, node), true);
+            if (!comp->codeGen->gcInfo.gcIsWriteBarrierAsgNode(node))
+            {
+                LowerStoreIndir(node->AsIndir());
+            }
             break;
 
         case GT_ADD:
-            return LowerAdd(node);
+        {
+            GenTree* afterTransform = LowerAdd(node);
+            if (afterTransform != nullptr)
+            {
+                return afterTransform;
+            }
+            __fallthrough;
+        }
```

Comment:
OK, I've moved the call to `ContainCheckFloatBinary()` into `ContainCheckBinary()` in lowerxarch.cpp. Looks a bit cleaner now.